### ISSUE TITLE
GetPropertyValue for EnabledState is crashing the plugin

### ISSUE THREAD

**Issue by mabrahamsen** [beyondcomputing-org/NewRelic.HyperV#4]
VMs, that are turned off and with checkpoints, has a null value for enabled state - and the GetPropertyValue is failing. Other offline machines without checkpoints work fine.

UPDATE: Seems this problem happens with checkpoints in general, and not tied to on or off.

I did a quick and dirty patch locally where I looped through the collection and checked for null, which works fine:
                UInt16 state = (value[0]
                    .Properties.Cast<PropertyData>()
                    .Where(property => property.Name == "EnabledState")
                    .Where(property => property.Value != null)
                    .Select(property => (UInt16) property.Value)).FirstOrDefault();

This was on Win2012R2.

Not sure what the proper solution is though.

**Comment by eelkram** [comment]
Thanks for the info and all the details, I'll take a look and see what I find.

**Comment by eelkram** [comment]
I've committed a fix that will address the issue with VM's that have checkpoints.  The way I was previously pulling VM data returned a collection of settings for a VM which included checkpoint settings data.  The new method is simpler and doesn't return the checkpoint settings, just the current vm settings.  This will allow fetching of CPU and Memory usage for VM's that are running and have a checkpoint.  I also added some logging and cleaned up a few things to make this easier to troubleshoot in the future.

I'll finish up packaging the plugin and push it out to New Relic tomorrow.
